app/models/repository.rb

```diff
@@ -28,8 +48,12 @@ def change_mirroring!(mirroring_enabled)
     update_column(:mirroring_enabled, mirroring_enabled)
   end
 
-  def self.remove_suse_repos_without_tokens!
-    where(auth_token: nil).where('external_url LIKE ?', 'https://updates.suse.com%').delete_all
+  def custom?
+    scc_id.nil?
+  end
+
+  def remove_if_custom
+    destroy! if custom?
```

Comment:
It's a very simple one-line method and it seems it is used only once. Is it really required?

Reply:
I would say that the extra protection in this method is better than having the possibility of the same logic being repeated and implemented incorrectly. This is our protection from deleting products from the database.

Reply:
I still can just call `Repository.destroy!` anywhere in the code or in the console. Does having this method really protect against anything?

Adding a destroy hook and raising an exception if the repo is non-custom would prevent this.

Reply:
But I think you are trying to solve the problem that doesn't exist/can't be solved. Other developers wouldn't know that this method exists, and would just call `destroy!` instead -- can't protect against that.

Reply:
Good idea, I just pushed a new commit with callback